fairreckitlib/dataloaders/dataloaders.py

```diff
@@ -111,128 +154,253 @@ def get_item_id(self, item_index: int) -> int:
         fields_map = {"gender": 1, "age": 2, "country": 3}
         use_cols = [0]
-        use_cols.extend([fields_map[key] for key in filters])
-        headers = ["user", "gender", "age", "country"]
+        use_cols.extend([fields_map[col_name] for col_name in add_columns])
         headers = [headers[i] for i in sorted(use_cols)]
         params = dict(delimiter=self.configs.get("common", "DELIMITER", fallback=","),
                       names=headers, engine='python', usecols=use_cols)
         data_frame = pd.read_csv(
             self.get_file_path(self.configs.get(self._dataset_name, 'file_path'),
                                file_name), **params)
+        self.ui_data_frame = pd.merge(self.ui_data_frame, data_frame, on=["user"], how="left")
         values = {"gender": "", "age": -1, "Country": ""}
-        data_frame.fillna(value=values, inplace=True)
-        df_filters = [data_frame[k].map(lambda x: str(x).lower()) == str(v).lower() if k != 'age'
-                      else data_frame[k].astype(int).between(int(filters[k][0]), int(filters[k][1]),
-                                                             inclusive = "both")
-                      for k, v in filters.items()]
-        data_frame = data_frame[ft.reduce(lambda x, y: (x) & (y), df_filters)]["user"]
-        data_frame =  pd.merge(self.ui_data_frame, data_frame, on=["user"], how="left")
-        self.ui_data_frame = data_frame
+        self.ui_data_frame.fillna(value=values, inplace=True)
 
-    def get_user_item_matrix(self, filters: Dict[str, Any] = None) -> pd.DataFrame:
-        """
-        Void function to make sparse user-item matrix
-        """
-        if not self.ui_data_frame:
-            self.load_data()
 
-        if filters:
-            self.filter_df(filters)
-
-        data_frame = self.ui_data_frame
-        data_frame["user_id"] = (data_frame['user']
-                                 .map(lambda x: zlib.adler32(str(x).encode('utf-8',
-                                                                           errors='ignore'))))
-        data_frame["item_id"] = (data_frame["item"]
-                                 .map(lambda x: zlib.adler32(str(x).encode('utf-8',
-                                                                            errors='ignore'))))
+class Dataloader1B(DataloaderBase):
+    """
+    In order to load 1B
+    """
 
-        users = data_frame["user_id"].unique()
-        items = data_frame["item_id"].unique()
-        shape = (len(users), len(items))
+    options = {"albums": ["album-id", "album-name", "artist-id"],
+               "artists": ["artist-id", "artist-name"],
+               "tracks": ["track-id", "track-name", "artist-id"],
+               "LEs": ["user-id", "artist-id", "album-id", "track-id", "timestamp"],
+               "users": ["user-id", "country", "age", "gender", "playcount",
+                         "registered_timestamp"],
+               "users_additional": ["user-id",
```

Comment:
what is this huge array and where is the comment explaining it?

Reply:
this is the headers of the file contains user_additional data . you can check it in the related paper.  Project Plan H - FairRecKit

Reply:
a different dev would not know that it would be in our project plan and would not have access to it, please add more comments explaining your code :)

Reply:
updated from
    http://www.cp.jku.at/people/schedl/Research/Publications/pdf/schedl_icmr_2016.pdf